shared/rakefiles/xk_util.rake

```diff
@@ -220,4 +220,35 @@ task :revoke_org_admin => [@gcp_creds_file, :configure_extra_tf_vars] do
   end
 end
 
+# This task restores a list of images in snapshots. 
+task :restore_snapshot_from_image_file, [:snapshot_files] => [@gcp_creds_file, :configure_extra_tf_vars] do |taskname, args|
+  require 'csv'
+
+  pv_zones = {}
+  CSV.parse(%x{
+              #{@exekube_cmd} kubectl get pv -o json | jq -r '.items[] | \"\\(.spec.claimRef.name),\\(.metadata.labels.\"failure-domain.beta.kubernetes.io/zone\")\"'
+           }.chomp).each do |line|
+             pv_zones[line[0]] = line[1]
```

Comment:
I don't understand what this stanza does. Why does it need to parse as CSV? Why is it replacing the 0th line with the 1st line? What does "line" mean?

Maybe some sample output would illustrate what this is trying to do.

Reply:
I added some comments in this commit to make this easier to understand: https://github.com/gpii-ops/gpii-infra/pull/432/commits/f19194f2a1e1cc6ec51cdd12aa9eac14bb7ff4ae

any advice is welcome :)